homework/shared_ptr/test.cpp

```diff
@@ -75,8 +82,25 @@ TEST_F(SharedPointerTestSuite, moveAssigmenOperator)
     my_shared_ptr<SomeUsefulClass> some_ptr = my_shared_ptr<SomeUsefulClass>(new SomeUsefulClass());;
     my_shared_ptr<SomeUsefulClass> another_ptr = my_shared_ptr<SomeUsefulClass>(new SomeUsefulClass());;
     another_ptr  = std::move(some_ptr);  // now we have move assignment
+
+    EXPECT_EQ(another_ptr->message_, "");
+    EXPECT_EQ(another_ptr.use_count(), 1);
+
+    EXPECT_EQ(some_ptr.get(), nullptr);
+    EXPECT_EQ(some_ptr.use_count(), 0);
+    ASSERT_FALSE(some_ptr);
+}
+
+TEST_F(SharedPointerTestSuite, moveConstructor)
+{
+    my_shared_ptr<SomeUsefulClass> some_ptr = my_shared_ptr<SomeUsefulClass>(new SomeUsefulClass());;
+    my_shared_ptr<SomeUsefulClass> another_ptr = my_shared_ptr<SomeUsefulClass>(std::move(some_ptr));;
```

Comment:
hmm... interesting code. You have used both copy and move constructors in one line :D 
Probably you have just meant to use only move constructor like below.
```suggestion
    my_shared_ptr<SomeUsefulClass> another_ptr = std::move(some_ptr);
```

Anyway, move c-tor was used :D